### ISSUE TITLE
DNDB - Cast Spell - Eldritch Blast roll expression does not include damage

### ISSUE THREAD

**Issue by trey-kirk-sp** [mr-ice/maptool-macros#87]
charId: 20170550

Cast Spell - Eldritch Blast - max level

Result:
```
[{
  "name": "Eldritch Blast",
  "bonus": 10,
  "diceSize": "20",
  "diceRolled": "1",
  "expressionTypes": "Attack",
  "description": "",
  "output": "Rolled: 2 Eldritch Blast Attack Roll: 1d20 + 10 = 12 ",
  "rolls": [2],
  "roll": 2,
  "rollString": "1d20 + 10",
  "total": 12,
  "totals": [12],
  "outputs": ["Rolled: 2 Eldritch Blast Attack Roll: 1d20 + 10 = 12 "],
  "allTotal": 12
}]
```
Class obj:
```
[
    {
    "className": "Bard",
    "level": 13,
    "hitDice": 8,
    "hitDiceUsed": 0
  },
    {
    "className": "Warlock",
    "level": 1,
    "hitDice": 8,
    "hitDiceUsed": 0
  }
]
```
Looks like the single bolt is correct (War 1), but 1d10 damage expression was expected.

**Comment by trey-kirk-sp** [comment]
Chill touch is also lacking:
```
[{
  "name": "Chill Touch",
  "bonus": 7,
  "diceSize": "20",
  "diceRolled": "1",
  "expressionTypes": "Attack",
  "description": "",
  "output": "Rolled: 10 Chill Touch Attack Roll: 1d20 + 7 = 17 ",
  "rolls": [10],
  "roll": 10,
  "rollString": "1d20 + 7",
  "total": 17,
  "totals": [17],
  "outputs": ["Rolled: 10 Chill Touch Attack Roll: 1d20 + 7 = 17 "],
  "allTotal": 17
}]
```

**Comment by trey-kirk-sp** [comment]
Not sure why there's no damage, but it's probably related to the atHigherLevels modifier. Eldritch Blast has the same hallmark of what's becoming a reoccurring theme: spells that when upcast do not increase damage or effect, but rather repeat the effect. Aka Magic Missile and Scorching Ray. The hallmark looks something like:
```
    "modifiers": [    {
      "atHigherLevels":       {
        "additionalAttacks": [],
        "additionalTargets": [],
        "areaOfEffect": [],
        "creatures": [],
        "duration": [],
        "higherLevelDefinitions": [],
        "points": [],
        "scaleType": "spellscale",
        "special": []
      }
```
Where the 'atHigherLevels' entry doesn't have a corresponding higherLevelsDefinition, which is what tells us how to modify the spell. Instead it's empty with the exception of the scaleType. For Magic Missile and Scorching Ray, it's spellscale which just indicates the higher spell slot spent, the more the effect is repeated.

Eldritch Blast is slightly different:
```
"modifiers": [    {
      "atHigherLevels":       {
        "additionalAttacks": [],
        "additionalTargets": [],
        "areaOfEffect": [],
        "creatures": [],
        "duration": [],
        "higherLevelDefinitions": [],
        "points": [],
        "scaleType": "characterlevel",
        "special": []
      }
```
scaleType characterlevel means the effect repeats as the character's level increases. The bad news in this case is that there's no structured information to help us determine at which level the effect is repeated. So at this point, the spell effect will not be repeated in the corresponding RollExpression. The user will just have to repeat the casting manually to pull that off. 

So all I can do right now is just fix it so the Damage expression is built.

**Comment by trey-kirk-sp** [comment]
Fixed in hash 34c46dd7
